src/AppInstallerCommonCore/Public/winget/Fonts.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#pragma once
+#include <string>
+#include <vector>
+
+namespace AppInstaller::Fonts
+{
+    struct FontFace
+    {
+        std::wstring Name;
+        std::vector<std::filesystem::path> FilePaths;
+
+        // TODO: Add support for font face versions.
+    };
+
+    struct FontFamily
+    {
+        std::wstring Name;
+        std::vector<FontFace> Faces;
+    };
+
+    /// <summary>
+    /// Gets all installed font families on the system.
+    /// </summary>
+    /// <returns>A list of installed font families.</returns>
+    std::vector<FontFamily> GetInstalledFontFamilies();
+
+    /// <summary>
+    /// Gets the installed font family from the provided family name.
+    /// </summary>
+    /// <param name="familyName">The font family name.</param>
+    /// <returns>The specified font family if found.</returns>
+    std::optional<FontFamily> GetInstalledFontFamily(const std::wstring& familyName);
```

Comment:
Seems a bit odd to me to have both a singular and a pluralized version of effectively the same function, with the only difference being that one takes a parameter to search for. Wouldn't it be possible to simply have `GetInstalledFontFamilies` with an optional parameter for the search string? Then if no families are found, you have an empty vector, if an exact match is found you'd have a vector with a single `FontFamily` in it, and if no `familyName` was specified, you'd have a vector of all the families.

Reply:
Thanks for the suggestion, I agree that there's a lot of repeated code here so I combined the functionality as suggested.